### PR TITLE
tools/tpm2_pcrread: add support for calculating cpHash

### PR DESCRIPTION
**Pull request by ccihula**
Signed-off-by: cihula <connor.cihula@intel.com>

### CODE REVIEW

File: lib/tpm2_policy.c [hunk 1]
```diff
@@ -144,7 +144,7 @@ tool_rc tpm2_policy_build_pcr(ESYS_CONTEXT *ectx, tpm2_session *policy_session,
         // Read PCRs
         tool_rc rc = tpm2_pcr_read(ectx, ESYS_TR_NONE, ESYS_TR_NONE,
                 ESYS_TR_NONE, pcr_selections, &pcr_update_counter,
-                NULL, &pcr_val);
+                NULL, &pcr_val, NULL, TPM2_ALG_NULL);
```
Comment (idesai) [inline lib/tpm2_policy.c:147]:
```suggestion
                NULL, &pcr_val, NULL, TPM2_ALG_ERROR);
```

File: tools/tpm2_quote.c [hunk 2]
```diff
@@ -158,7 +158,8 @@ static tool_rc process_output(ESYS_CONTEXT *ectx) {
         }
 
         // Gather PCR values from the TPM (the quote doesn't have them!)
-        rc = pcr_read_pcr_values(ectx, &ctx.pcr_selections, &ctx.pcrs);
+        rc = pcr_read_pcr_values(ectx, &ctx.pcr_selections, &ctx.pcrs,
+            &ctx.cp_hash, ctx.parameter_hash_algorithm);
```
Comment (idesai) [inline tools/tpm2_quote.c:162]:
```suggestion
            NULL, TPM2_ALG_ERROR);
```

Comment (idesai) [reply tools/tpm2_quote.c:162]: A call to pcr_read with the same variable name will clobber cpHash for tpm2_quote.

File: lib/pcr.c [hunk 3]
```diff
@@ -564,9 +565,9 @@ tool_rc pcr_read_pcr_values(ESYS_CONTEXT *esys_context,
         TPML_DIGEST *v;
         tool_rc rc = tpm2_pcr_read(esys_context, ESYS_TR_NONE, ESYS_TR_NONE,
                 ESYS_TR_NONE, &pcr_selection_tmp, &pcr_update_counter,
-                &pcr_selection_out, &v);
+                &pcr_selection_out, &v, cp_hash, parameter_hash_algorithm);
 
-        if (rc != tool_rc_success) {
+        if (rc != tool_rc_success || cp_hash->size) {
```
Comment (idesai) [inline lib/pcr.c:570]:
```suggestion
        if (rc != tool_rc_success || (cp_hash && cp_hash->size)) {
```

Comment (idesai) [reply lib/pcr.c:570]: we also need to test if cp_hash is valid, else we'd accidentally dereference a null pointer.

**Review by idesai — CHANGES_REQUESTED** [review]

**Review by idesai — APPROVED** [review]
